lib/AST/ASTPrinter.cpp

```diff
@@ -6143,7 +6143,10 @@ class TypePrinter : public TypeVisitor<TypePrinter, void, NonRecursivePrintOptio
       parent = alias->getParent();
     else if (auto generic = T->getAs<AnyGenericType>())
       parent = generic->getParent();
-    return parent && parent->isTypeParameter();
+    else if (auto dmt = T->getAs<DependentMemberType>())
+      parent = dmt->getBase();
```

Comment:
The base of a DependentMemberType should always be a type parameter, so you can just return `true` here without recursing